### PR TITLE
chore: added article creation e2e test

### PR DESCRIPTION
**Pull request by JohnAllenTech**
# ✨ Codu Pull Request 💻

Fixes #(issue)

Adds to #468 

image

## Pull Request details

- Authenticated E2E user writes a basic article and ensures that it can be published on mobile and desktop
- Checks for bookmark and like icon on article page
- Checks author is correct

## Any Breaking changes

- None

## Associated Screenshots

image
Generated article

### CODE REVIEW

File: e2e/articles.spec.ts [hunk 1]
```diff
@@ -124,4 +124,55 @@ test.describe("Authenticated Articles Page", () => {
...
+      .getByPlaceholder("Enter your content here 💖")
+      .fill(articleContent);
+
+    await expect(page.getByRole("button", { name: "Next" })).toBeVisible();
+    await page.getByRole("button", { name: "Next" }).click();
+    await page.waitForTimeout(1000);
+    await page.getByRole("button", { name: "Publish now" }).click();
+    await page.waitForTimeout(1000);
+    await page.waitForURL(
+      /^http:\/\/localhost:3000\/articles\/lorem-ipsum-.*$/,
```
[the remark this answers is not in the document]
Comment (JohnAllenTech) [reply e2e/articles.spec.ts]: Good catch. I will do this

File: e2e/articles.spec.ts [hunk 2]
```diff
@@ -124,4 +124,55 @@
...
+      await expect(
+        page.getByRole("button", { name: "Open main menu" }),
+      ).toBeVisible();
+      page.getByRole("button", { name: "Open main menu" }).tap();
+      await expect(page.getByRole("link", { name: "New Post" })).toBeVisible();
+      await page.getByRole("link", { name: "New Post" }).tap();
+    } else {
+      await expect(page.getByRole("link", { name: "New Post" })).toBeVisible();
+      await page.getByRole("link", { name: "New Post" }).click();
+    }
```
[the remark this answers is not in the document]
Comment (JohnAllenTech) [reply e2e/articles.spec.ts:147]: Not until its reused

File: e2e/articles.spec.ts [hunk 3]
```diff
@@ -124,4 +124,55 @@
...
+    await page.getByPlaceholder("Article title").fill(articleTitle);
+
+    await page
+      .getByPlaceholder("Enter your content here 💖")
+      .fill(articleContent);
+
+    await expect(page.getByRole("button", { name: "Next" })).toBeVisible();
+    await page.getByRole("button", { name: "Next" }).click();
+    await page.waitForTimeout(1000);
+    await page.getByRole("button", { name: "Publish now" }).click();
```
[the remark this answers is not in the document]
Comment (JohnAllenTech) [reply e2e/articles.spec.ts]: No thats a terrible idea. 'networkidle' is not guaranteed as there is background requests happening

File: e2e/articles.spec.ts [hunk 4]
```diff
@@ -124,4 +124,55 @@
...
+      ).toBeVisible();
+      page.getByRole("button", { name: "Open main menu" }).tap();
+      await expect(page.getByRole("link", { name: "New Post" })).toBeVisible();
+      await page.getByRole("link", { name: "New Post" }).tap();
+    } else {
+      await expect(page.getByRole("link", { name: "New Post" })).toBeVisible();
+      await page.getByRole("link", { name: "New Post" }).click();
+    }
+    await page.waitForTimeout(1000);
+
```
[the remark this answers is not in the document]
Comment (JohnAllenTech) [reply e2e/articles.spec.ts]: Done

**Review by NiallJoeMaher — APPROVED** [review]
🌮🌮🌮 LGTM!

**Pull request merged**